js/modules/k6/metrics/metrics.go

```diff
@@ -52,14 +53,17 @@ type Metric struct {
 }
 
 // ErrMetricsAddInInitContext is error returned when adding to metric is done in the init context
-var ErrMetricsAddInInitContext = common.NewInitContextError("Adding to metrics in the init context is not supported")
+var (
+	ErrMetricsAddInInitContext = common.NewInitContextError("Adding to metrics in the init context is not supported")
```

Comment:
Raising exceptions that interrupt the execution makes the tool more fragile. 
Errors are expected when doing stress testing, so I think raising exceptions is not a good default because it will condition users to wrap code in `try/catch`. 
I think k6 should be resilient to errors, as I argued in  https://github.com/loadimpact/k6/issues/1435

Reply:
Given that:

1. this will mean that they will just be getting spammed by log messages until #1681, without a way to stop them
2. this is practically only reported by you, so I would argue it is fairly rare? :thinking: and as such IMO an actual "exception" :wink: :wink:   

I think this should throw an exception with a stacktrace instead of just logging  `Adding to metrics in the init context is not supported` and letting the user guess where that is and what should they do about it. 

I guess we can add the stacktrace to the warning but that IMO will be even noisier.